### PR TITLE
add frontend guidelines

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does

Add frontend guidelines as md file

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: grafana-plugin/frontend_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# OnCall Frontend Guidelines
+
+## Ways of working
+
+### Code reviews
+
+- PR can be merged when the following conditions are met:
+   pipeline succeeded without bypassing checks
+  ere is at least 1 approval from the frontend team and no opened remarks from reviewers other than the approver
```
Comment (joeyorlando) [inline grafana-plugin/frontend_guidelines.md:9]:
```suggestion
  there is at least 1 approval from the frontend team and no opened remarks from reviewers other than the approver
```

File: grafana-plugin/frontend_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  l comments are replied, answered, addressed or discussed separately
```
Comment (joeyorlando) [inline grafana-plugin/frontend_guidelines.md:10]:
```suggestion
  l comments are replied, answered, addressed or discussed separately
```

the "l" here ☝️  is a typo maybe?

File: grafana-plugin/frontend_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+   make use of builtin GH PR reviews
+  nor comments that don’t need to be addressed right away and are not a blocker for a merge are marked with “Minor: “ prefix
+  nless there is some important hotfix to make and no frontend mates are available*
```
Comment (joeyorlando) [inline grafana-plugin/frontend_guidelines.md:13]:
```suggestion
  unless there is some important hotfix to make and no frontend mates are available*
```

File: grafana-plugin/frontend_guidelines.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+### Use global decorators / custom hooks / utilities consistently
+
+- For example:
+  ithGlobalNotification to set successful / failing notification based on MobX action’s result
+  utoLoadingState to manage loading statuses of async actions
+  eIsLoading to consume loading state
+  eDrawer to manage drawers
+  eConfirmModal to manage confirm modal
+  obal helpers
```
Comment (joeyorlando) [inline grafana-plugin/frontend_guidelines.md:59]: seems odd here, looks like the first letter of every line got cut off?

**Review by joeyorlando — APPROVED** [review]
great initiative 🫶 

did you consider putting this under the `dev` root directory?

File: grafana-plugin/frontend_guidelines.md (hunk 4 above)
Comment (brojd) [reply grafana-plugin/frontend_guidelines.md]: It was broken by copy pasting from markdown previewer tool, I fixed all the places now, thanks

**Comment by brojd** [comment]
@joeyorlando I moved it to /dev folder

**Review by joeyorlando — APPROVED** [review]

**Review by teodosii — APPROVED** [review]

**Pull request merged**